rero_mef/authorities/gnd/tasks.py

```diff
@@ -0,0 +1,250 @@
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Tasks used by  RERO-MEF."""
+
+import datetime
+from datetime import timedelta
+from io import StringIO
+
+import click
+from celery import shared_task
+from dateutil import parser
+from flask import current_app
+from invenio_db import db
+from invenio_oaiharvester.api import get_info_by_oai_name
+from invenio_oaiharvester.errors import WrongDateCombination
+from invenio_oaiharvester.utils import get_oaiharvest_object
+from pymarc.marcxml import parse_xml_to_array
+from sickle import Sickle, oaiexceptions
+from sickle.iterator import OAIItemIterator
+from sickle.oaiexceptions import NoRecordsMatch
+
+from .api import GndRecord
+from ..marctojson.do_gnd_auth_person import Transformation
+from ..tasks import get_record
+
+
+class MyOAIItemIterator(OAIItemIterator):
+    """OAI item iterator with accessToken."""
+
+    def _next_response(self):
+        """Get the next response from the OAI server."""
+        params = self.params
+        access_token = params.get('accessToken')
+        if self.resumption_token:
+            params = {
+                'resumptionToken': self.resumption_token.token,
+                'verb': self.verb
+            }
+        if access_token:
+            params['accessToken'] = access_token
+
+        self.oai_response = self.sickle.harvest(**params)
+        error = self.oai_response.xml.find(
+            './/' + self.sickle.oai_namespace + 'error')
+        if error is not None:
+            code = error.attrib.get('code', 'UNKNOWN')
+            description = error.text or ''
+            try:
+                raise getattr(
+                    oaiexceptions, code[0].upper() + code[1:])(description)
+            except AttributeError:
+                raise oaiexceptions.OAIError(description)
+        if self.resumption_token:
+            # Test we got a complete response ('resumptionToken' in xml)
+            resumption_token_element = self.oai_response.xml.find(
+                './/' + self.sickle.oai_namespace + 'resumptionToken')
+
+            if resumption_token_element is None:
+                msg = ('ERROR HARVESTING incomplete response:'
+                       '{cursor} {token}').format(
+                    cursor=self.resumption_token.cursor,
+                    token=self.resumption_token.token
+                )
+                current_app.logger.error(msg)
+            else:
+                self.resumption_token = self._get_resumption_token()
+
+                self._items = self.oai_response.xml.iterfind(
+                    './/' + self.sickle.oai_namespace + self.element)
+
+        else:
+            # first time
+            self.resumption_token = self._get_resumption_token()
+
+            self._items = self.oai_response.xml.iterfind(
+                './/' + self.sickle.oai_namespace + self.element)
```

Comment:
Both statements contain same code. 
If more lines are necessary in the future, maybe it's better to extract these line into a separate inner method.

Reply:
ok